src/sentry/static/sentry/app/views/organizationIntegrations/constants.tsx

```diff
@@ -2,8 +2,56 @@ export const INSTALLED = 'Installed' as const;
 export const NOT_INSTALLED = 'Not Installed' as const;
 export const PENDING = 'Pending' as const;
 
-export const colors = {
+export const COLORS = {
   [INSTALLED]: 'success',
   [NOT_INSTALLED]: 'gray2',
   [PENDING]: 'yellowOrange',
 };
+
+/**
+ * Integrations in the integration directory should be sorted by their popularity (weight).
+ * These weights should be hardcoded in the application itself.
+ * We can store this in a map where the key is the integration slug and the value is an integer and represents the weight.
+ * The weights should reflect the relative popularity of each integration.
+ */
+
+export const POPULARITY_WEIGHT: {
+  [key: string]: number;
```

Comment:
I feel like this can be narrowed into `Integration['name']` and the accepted value for `Integration['name']` should be narrowed into `'slack' | 'github' | ...` but that's for another PR.

If you're using `{ [key: string]: number; }` then would it make sense to make it readonly with `as const` at the end?

Reply:
@leedongwei currently Integration['name'] is just typed as string. But yes, makes sense to add `as const`.

Reply:
Y'all should consider making the type for `Integration` be stricter.

Reply:
@leedongwei it's a complex issue b/c of the the different types of integrations

Reply:
If it is complex, isn't that a stronger reason to get the types correct and true so that the typechecker can support the next developer working on this?

I think it's fine to say that you're making a trade-off on correctness vs speed.

Reply:
@leedongwei lol these typings were made before me and used everywhere. It's out of the scope to refactor in this PR. 

I'm definitely making the trade-off in favour of speed 😄